### PR TITLE
add pipeline distributed training test

### PR DESCRIPTION
**Pull request by xzhu1900**
**Description**: add pipeline distributed training test to night CI. The test currently is running on a machine with 4 GPUs. So 4-way PP and 2-way DP+2-way PP are tested.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Review by liqunfu — DISMISSED** [review]

File: tools/ci_build/github/linux/run_dockerbuild.sh [hunk 1]
```diff
@@ -128,6 +128,11 @@ if [[ $BUILD_EXTR_PAR = *--enable_training_python_frontend_e2e_tests* ]]; then
     # DOCKER_RUN_PARAMETER="$DOCKER_RUN_PARAMETER -u0"
 fi
 
+if [[ $BUILD_EXTR_PAR = *--enable_training_pipeline_e2e_tests* ]]; then
+    DOCKER_RUN_PARAMETER="$DOCKER_RUN_PARAMETER --volume /bert_ort:/bert_ort \
```
Comment (edgchen1) [inline tools/ci_build/github/linux/run_dockerbuild.sh:132]: is it possible to avoid hardcoding the host path?

File: tools/ci_build/build.py [hunk 2]
```diff
@@ -1134,6 +1137,51 @@ def run_training_python_frontend_e2e_tests(cwd):
         'BertModelTest.test_for_pretraining_mixed_precision_with_gradient_accumulation'], cwd=cwd)
 
 
+def run_training_pipeline_e2e_tests(cwd):
```
Comment (edgchen1) [inline tools/ci_build/build.py:1140]: adding these to build.py is one way to do it. however, i think there's already a lot here. it's also possible to run tests from separate scripts, as in https://github.com/microsoft/onnxruntime/blob/master/tools/ci_build/github/azure-pipelines/orttraining-linux-gpu-e2e-test-ci-pipeline.yml

File: tools/ci_build/github/linux/run_dockerbuild.sh (hunk 1 above)
Comment (xzhu1900) [reply tools/ci_build/github/linux/run_dockerbuild.sh:132]: I think it is fine using hard code path here: the path is a well established place for model accessing and is being used in other tests for model accessing too. For example: orttraining-linux-gpu-perf-test-ci-pipeline.yml

File: tools/ci_build/build.py (hunk 2 above)
Comment (xzhu1900) [reply tools/ci_build/build.py:1140]: I'd prefer using build.py as it has some mpi config code I can re-use in my model parallelism test. This file also contains other distributed test so kind of keeping the test centralized.

**Review by liqunfu — APPROVED** [review]

**Pull request merged**